test/unit/mdc-line-ripple/mdc-line-ripple-foundation.test.js

```diff
@@ -56,20 +56,62 @@ test('#destroy removes event listeners', () => {
   td.verify(mockAdapter.deregisterEventHandler('transitionend', td.matchers.isA(Function)));
 });
 
-test(`activate adds ${MDCLineRippleFoundation.cssClasses.LINE_RIPPLE_ACTIVE} class`, () => {
+test(`activate adds ${cssClasses.LINE_RIPPLE_ACTIVE} class`, () => {
   const {foundation, mockAdapter} = setupTest();
   foundation.init();
   foundation.activate();
   td.verify(mockAdapter.addClass(cssClasses.LINE_RIPPLE_ACTIVE));
 });
 
-test(`deactivate removes ${MDCLineRippleFoundation.cssClasses.LINE_RIPPLE_DEACTIVATING} class`, () => {
+test(`deactivate adds ${cssClasses.LINE_RIPPLE_DEACTIVATING} class`, () => {
   const {foundation, mockAdapter} = setupTest();
+  td.when(mockAdapter.hasClass(cssClasses.LINE_RIPPLE_DEACTIVATING)).thenReturn(true);
+
   foundation.init();
   foundation.deactivate();
   td.verify(mockAdapter.addClass(cssClasses.LINE_RIPPLE_DEACTIVATING));
 });
 
+test('opacity event after adding deactivating class triggers activation classes to be removed', () => {
+  const {foundation, mockAdapter} = setupTest();
+  td.when(mockAdapter.hasClass(cssClasses.LINE_RIPPLE_DEACTIVATING)).thenReturn(true);
+  const event = {
+    propertyName: 'opacity',
+  };
+
+  foundation.init();
+  foundation.deactivate();
+  td.verify(mockAdapter.addClass(cssClasses.LINE_RIPPLE_DEACTIVATING));
```

Comment:
This test seems like 2 in 1. 

One test can just be, "#deactivate adds the cssClasses.LINE_RIPPLE_DEACTIVATING class". The other can be opacity event triggers ...